src/main/java/com/projectkorra/core/util/ParticleUtil.java

```diff
@@ -0,0 +1,603 @@
+	}
+	
+	/**
+	 * Spawn a single FALLING_DUST particle with the given BlockData at the given Location
+	 * @param block BlockData for the FALLING_DUST particle to show
+	 * @param loc where to spawn the particle
+	 */
+	public static void spawnFallingDust(BlockData block, Location loc) {
+		spawnFallingDust(block, loc, 1);
+	}
+	
+	/**
+	 * Spawn an amount of the ITEM_CRACK particle with the given ItemStack at the given location within the specified bounds
+	 * @param item ItemStack for the ITEM_CRACK particle to show
+	 * @param loc where to spawn the particle
+	 * @param amount how many of the particle to spawn
+	 * @param offsetX x-axis offset bounds for randomized spawning
+	 * @param offsetY y-axis offset bounds for randomized spawning
+	 * @param offsetZ z-axis offset bounds for randomized spawning
+	 */
+	public static void spawnItemCrack(ItemStack item, Location loc, int amount, double offsetX, double offsetY, double offsetZ) {
+		if (item == null) {
+			item = new ItemStack(Material.BARRIER);
+		}
+		
+		spawn(Particle.ITEM_CRACK, loc, amount, offsetX, offsetY, offsetZ, 0, item);
+	}
+	
+	/**
+	 * Spawn an amount of the ITEM_CRACK particle with the given ItemStack at the given Location within the specified bounds
+	 * @param item ItemStack for the ITEM_CRACK particle to show
+	 * @param loc where to spawn the particle
+	 * @param amount how many of the particle to spawn
+	 * @param offsets offset bounds for equal randomized spawning on each axis
+	 */
+	public static void spawnItemCrack(ItemStack item, Location loc, int amount, double offsets) {
+		spawnItemCrack(item, loc, amount, offsets, offsets, offsets);
+	}
+	
+	/**
+	 * Spawn an amount of the ITEM_CRACK particle with the given ItemStack at the given Location
+	 * @param item ItemStack for the ITEM_CRACK particle to show
+	 * @param loc where to spawn the particle
+	 * @param amount how many of the particle to spawn
+	 */
+	public static void spawnItemCrack(ItemStack item, Location loc, int amount) {
+		spawnItemCrack(item, loc, amount, 0);
+	}
+	
+	/**
+	 * Spawn a single ITEM_CRACK particle with the given ItemStack at the given Location
+	 * @param item ItemStack for the ITEM_CRACK particle to show
+	 * @param loc where to spawn the particle
+	 */
+	public static void spawnItemCrack(ItemStack item, Location loc) {
+		spawnItemCrack(item, loc, 1);
+	}
+	
+	/**
+	 * Spawn an amount of colored NOTE particles at the given Location within the specified bounds
+	 * @param color which color the note should be
+	 * @param loc where to spawn the particle
+	 * @param amount how many of the particle to spawn
+	 * @param offsetX x-axis offset bounds for randomized spawning
+	 * @param offsetY y-axis offset bounds for randomized spawning
+	 * @param offsetZ z-axis offset bounds for randomized spawning
+	 */
+	public static void spawnColoredNote(NoteColor color, Location loc, int amount, double offsetX, double offsetY, double offsetZ) {
+		World world = loc.getWorld();
```

Comment:
why do you create an local instance of World here instead of just doing loc.getWorld()

Reply:
Probably to prevent calling the accessor in the loop on line 287. It seems to be the same in `spawnColoredSpell`.

Reply:
Yeah but its not the same in spawn() i was just wondering why it wasn't consistent

Reply:
`spawn` passes the amount directly into the Spigot display function. Here, he has to manually spawn each particle because setting the amount to 0 is necessary to give it a color.

Reply:
Ahhhh, didn't know that